Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/dashboard.js

```diff
@@ -348,7 +348,28 @@ class Dashboard {
         if (!this.ref.state.displayed || !this.requestData)
             return;
 
-        const now = Math.floor(Date.now() / 1000);
+        const globalParams = this.documentParams()
+        let now = Math.floor(Date.now() / 1000);
+        if (Object.hasOwn(globalParams, 'before_time'))
+            now = Math.min(now, globalParams.before_time[0]);
+        if (Object.hasOwn(globalParams, 'before_timestamp'))
+            now = Math.min(now, globalPaarams.before_timestamp[0]);
+        if (Object.hasOwn(globalParams, 'before_uuid'))
+            now = Math.min(now, Math.floor(globalParams.before_uuid[0] / 100));
+        if (Object.hasOwn(globalParams, 'before_ref')) {
+            let commit_banck = CommitBank;
+            if (Object.hasOwn(globalParams, 'branch'))
+                this.commit_bank = CommitBank.forBranch(globalParams.branch);
```

Comment:
`this.commit_bank` also doesn't seem to be used? Perhaps I don't understand the interface. Are we supposed to store it somewhere, but then use class singleton methods (e.g. `CommitBank.commits`) to access commit refs?

Reply:
This code should be `commit_bank = ...`.

`CommitBank` is a singleton class, but we need specific instances for each branch, and it's possible for us to be ranged based on a commit which isn't on the branch associated with the specific dashboard we're dealing with. My testing did a bad job at reasoning about this case because these subtleties don't matter much when you're on `main`.